Sprinkle of Yeezus/Settings.swift

```diff
@@ -11,52 +11,68 @@ import UIKit
 import UserNotifications
 
 class SettingsPage: UIViewController {
-    
 
-    @IBOutlet weak var InfoLabel: UILabel!
-    @IBOutlet weak var DescButton: UIButton!
-    @IBOutlet weak var DescLabel: UILabel!
-    @IBOutlet weak var Switch: UISwitch!
+    @IBOutlet private weak var InfoLabel: UILabel!
+    @IBOutlet private weak var DescButton: UIButton!
+    @IBOutlet private weak var DescLabel: UILabel!
+    @IBOutlet private weak var Switch: UISwitch!
+    @IBOutlet private weak var TimePicked: UIDatePicker!
     
-    var notificationsOn = Bool()
+    private var sprinkleTimePicked = Date()
+    private let notificationCenter = UNUserNotificationCenter.current()
     
-    @IBAction func NotificationSwitch(_ sender: UISwitch) {
-        if Switch.isOn == true{
-            UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .sound], completionHandler: {didAllow, error in})
-            
-            DescLabel.isHidden = false
-            DescButton.isHidden = false
-            
-            notificationsOn = true
-        } else {
-            DescLabel.isHidden = true
-            DescButton.isHidden = true
-            notificationsOn = false
-        }
+    // MARK: - Lifecycle
+    override func viewDidLoad() {
+        sprinkleNotifications(sprinkleList)
     }
     
-    @IBOutlet weak var TimePicked: UIDatePicker!
-    var sprinkleTimePicked = Date()
-    @IBAction func TimePicker(_ sender: UIDatePicker){
-        sprinkleTimePicked = TimePicked.date
+    // MARK: - IBAction
+    @IBAction private func NotificationSwitch(_ sender: UISwitch) {
+        if sender.isOn {
+            notificationCenter.requestAuthorization(options: [.alert, .sound], completionHandler: { didAllow, error in
+                DispatchQueue.main.async {
+                    sender.isOn = didAllow
+                    self.updateUI()
+                }
+            })
+        }
+        
+        updateUI()
     }
     
-    func sprinkleNotifications(_ sprinkleList: Array<Sprinkle>) {
-        if notificationsOn == true{
-            
-            let notification = UNMutableNotificationContent()
-            let notificationText = pickRandomQuote(sprinkleList)
-            notification.body = " \"\(notificationText.quote)\" \(notificationText.quoteSource), \(notificationText.date)"
-            
-            let timeForSprinkle = TimePicked.date
-            let components = Calendar.current.dateComponents([.hour, .minute], from: timeForSprinkle)
-            InfoLabel.text = "Sprinkles will be sent every day at \(components.hour):\(components.minute)"
-            let notificationTime = UNCalendarNotificationTrigger(dateMatching: components, repeats: true)
-            
-        }
+    @IBAction private func TimePicker(_ sender: UIDatePicker){
+        sprinkleTimePicked = sender.date
+    }  
+    
+    // MARK: - Private
+    private func updateUI() {
+        [DescLabel, DescButton].forEach { $0.isHidden = !Switch.isOn }
```

Comment:
I don't know if you've seen the `forEach` function before, but it operates just like a `for` loop.

Basically, I made an array that contains the `DescLabel` and `DescButton` elements, then I call the `forEach` function not the array, which iterates over the array's objects and performs the change within the curly braces `{ }`.

`$0` is a syntactical way of referring to each object as you iterate. And so I set the `isHidden` value of each element to `!Switch.isOn` because we want the elements to be hidden is notifications are NOT enabled